lv2/sfizz.c

```diff
@@ -534,23 +534,43 @@ sfizz_lv2_status_log(sfizz_plugin_t *self)
     // lv2_log_note(&self->logger, "[sfizz] Active voices: %d\n", sfizz_get_num_active_voices(self->synth));
 }
 
+static int next_pow_2(int v)
+{
+    if (v < 1)
+        return 1;
+
+    // Bit twiddling hack
+    v--;
+    v |= v >> 1;
+    v |= v >> 2;
+    v |= v >> 4;
+    v |= v >> 8;
+    v |= v >> 16;
+    v++;
+    return v;
+}
+
 static void
 sfizz_lv2_check_oversampling(sfizz_plugin_t* self)
 {
-    sfizz_oversampling_factor_t oversampling = (sfizz_oversampling_factor_t)*self->oversampling_port;
-    if (oversampling != self->oversampling)
+    if (*self->oversampling_port < 0)
```

Comment:
Maybe apply next-pow-2 first on the port value, then compare?
and this makes this condition superfluous, given the negative cases is already handled in next-pow-2?

Reply:
Yeah I was a bit wary since well, it looks like more operations but I guess it's not a big problem.